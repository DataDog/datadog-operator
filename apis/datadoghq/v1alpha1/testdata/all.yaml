apiVersion: datadoghq.com/v1alpha1
kind: DatadogAgent
metadata:
  name: datadog
spec:
  clusterName: foo
  registry: public.ecr.aws/datadog
  credentials:
    apiKey: api-key-inline
    apiKeyExistingSecret: api-key-exist-secret
    apiSecret:
      secretName: datadog-secret
      keyName: api-key
    appKey: app-key-inline
    appKeyExistingSecret: app-key-existing-secret
    appSecret:
      secretName: datadog-secret
      keyName: app-key
    token: "foo-bar-baz" # ignored by conversion
    useSecretBackend: true
  features:
    orchestratorExplorer:
      enabled: true
      conf:
        configMap:
          name: orch-cm
          fileKey: config.yaml
      clusterCheck: true
      scrubbing:
        containers: true
      additionalEndpoints: '{"https://process.agent.datadoghq.com": ["apikey1"]}' # ignored ATM
      ddUrl: https://orch-explorer.com
      extraTags:
        - orch
        - exp
    kubeStateMetricsCore:
      enabled: true
      conf:
        configMap:
          name: cluster-agent-ksm-conf
      clusterCheck: true
    prometheusScrape:
      enabled: true
      serviceEndpoints: true
      additionalConfigs: |-
        - autodiscovery:
            kubernetes_annotations:
              include:
                custom_label: 'true'
            kubernetes_container_names:
            - my-app
          configurations:
          - send_distribution_buckets: true
            timeout: 5
    networkMonitoring:
      enabled: false
  agent:
    enabled: true
    useExtendedDaemonset: true #ignored ATM
    image:
      name: myagent
      tag: mytag
      jmxEnabled: true
      pullPolicy: Always
      pullSecrets: null
    daemonsetName: test-ds-agent
    rbac:
      create: true
      serviceAccountName: datadog-agent-scc
    deploymentStrategy:
      updateStrategyType: RollingUpdate
      rollingUpdate:
        maxUnavailable: 10
        maxPodSchedulerFailure: 10
        maxParallelPodCreation: 1
        slowStartIntervalDuration: 2h
        slowStartAdditiveIncrease: 1h
      canary: null
      reconcileFrequency: 1s
    additionalAnnotations:
      foo: bar
      baz: foo
    additionalLabels:
      bar: foo
      foo: baz
    priorityClassName: my-prio-class
    dnsPolicy: None
    dnsConfig:
      nameservers:
        - ns1
    hostNetwork: false
    hostPID: true
    env:
      - name: TEST_VAR_DS
        value: TEST_VALUE_DS
    config:
      ddUrl: custom-agent.com
      logLevel: DEBUG-LL # ignored
      confd:
        configMapName: agent-confd
      checksd:
        configMapName: agent-checksd
      podLabelsAsTags:
        custom-tag1: custom-tag1
      podAnnotationsAsTags:
        custom-tag2: custom-tag2
      tags:
        - hostTag1
        - hostTag2
      collectEvents: true
      leaderElection: true #ignored
      env:
        - name: TEST_VAR_CONT
          value: TEST_VALUE_CONT
      volumeMounts:
        - name: agent-volumeMount
      volumes:
        - name: agent-volume
      resources:
        limits:
          cpu: 400m
      command: ["custom-agent", "start"]
      args: ["config", "foo.yaml"]
      livenessProbe:
        initialDelaySeconds: 30
      readinessProbe:
        timeoutSeconds: 10
      healthPort: 42
      dogstatsd:
        dogstatsdOriginDetection: true
        unixDomainSocket:
          enabled: true
      tolerations:
        - operator: Exists
      securityContext:
        runAsUser: 0
        seLinuxOptions:
          level: s0
          role: system_r
          type: spc_t
          user: system_u
      criSocket:
        criSocketPath: /var/run/crio/crio.sock
        dockerSocketPath: /var/docker.sock
      hostPort: 420
      kubelet:
        tlsVerify: false
    apm:
      enabled: true
      unixDomainSocket:
        enabled: true
      hostPort: 9000
    process:
      enabled: true
      processCollectionEnabled: true
    log:
      enabled: true
    systemProbe:
      bpfDebugEnabled: true
      secCompCustomProfileConfigMap: seccomp-configmap
      secCompProfileName: localhost/seccomp-profile
      secCompRootPath: /custom/root/path
    security:
      compliance:
        enabled: true
      runtime:
        enabled: false
  clusterAgent:
    replicas: 2
    config:
      securityContext:
        runAsUser: 0
        seLinuxOptions:
          level: s0
          role: system_r
          type: spc_t
          user: system_u
      externalMetrics:
        enabled: true
        clusterChecksEnabled: true
      admissionController:
        enabled: true
        mutateUnlabelled: true
        serviceName: "test-svc-name"
        agentCommunicationMode: "hostip"
      confd:
        configMapName: cluster-agent-confd
        items:
          - key: test
            path: test.d/test.yaml
  clusterChecksRunner:
    enabled: true
    image:
      name: myagent
      tag: mytag
      jmxEnabled: true
      pullPolicy: Always
      pullSecrets: null
    deploymentName: "test-clc-name"
    config:
      securityContext:
        runAsUser: 0
        seLinuxOptions:
          level: s0
          role: system_r
          type: spc_t
          user: system_u
    customConfig:
      configData: "custom-conf-clc"
    rbac:
      create: true
      serviceAccountName: datadog-agent-clc
    replicas: 10
    additionalAnnotations:
      foo: bar
      baz: foo
    additionalLabels:
      bar: foo
      foo: baz
    priorityClassName: clc-prio-class
    affinity:
      nodeAffinity:
        requiredDuringSchedulingIgnoredDuringExecution:
          nodeSelectorTerms:
          - matchExpressions:
            - key: kubernetes.io/os
              operator: In
              values:
              - linux
    tolerations:
      - operator: Foo
    nodeSelector:
      nodeName: clc
    networkPolicy:
      create: true
      flavor: kubernetes
      dnsSelectorEndpoints:
        - matchLabels:
          component: redis
