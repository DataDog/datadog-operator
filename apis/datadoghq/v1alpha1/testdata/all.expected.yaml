metadata:
  creationTimestamp: null
  name: datadog
spec:
  features:
    admissionController:
      agentCommunicationMode: hostip
      enabled: true
      mutateUnlabelled: true
      serviceName: test-svc-name
    apm:
      enabled: true
      hostPortConfig:
        enabled: true
        hostPort: 9000
      unixDomainSocketConfig:
        enabled: true
    clusterChecks:
      useClusterChecksRunners: true
    cspm:
      enabled: true
    cws:
      enabled: false
    dogstatsd:
      originDetectionEnabled: true
      unixDomainSocketConfig:
        enabled: true
    eventCollection:
      collectKubernetesEvents: true
    externalMetricsServer:
      enabled: true
      useDatadogMetrics: false
    kubeStateMetricsCore:
      conf:
        configMap:
          name: cluster-agent-ksm-conf
      enabled: true
    liveContainerCollection:
      enabled: true
    liveProcessCollection:
      enabled: true
    logCollection:
      enabled: true
    npm:
      enabled: false
    orchestratorExplorer:
      conf:
        configMap:
          items:
          - key: config.yaml
            path: ""
          name: orch-cm
      enabled: true
      endpoint:
        url: https://orch-explorer.com
      extraTags:
      - orch
      - exp
      scrubContainers: true
    prometheusScrape:
      additionalConfigs: |-
        - autodiscovery:
            kubernetes_annotations:
              include:
                custom_label: 'true'
            kubernetes_container_names:
            - my-app
          configurations:
          - send_distribution_buckets: true
            timeout: 5
      enableServiceEndpoints: true
      enabled: true
  global:
    clusterName: foo
    credentials:
      apiKey: api-key-inline
      apiSecret:
        keyName: api-key
        secretName: datadog-secret
      appKey: app-key-inline
      appSecret:
        keyName: app-key
        secretName: datadog-secret
    criSocketPath: /var/run/crio/crio.sock
    dockerSocketPath: /var/docker.sock
    endpoint:
      url: custom-agent.com
    kubelet:
      tlsVerify: false
    registry: public.ecr.aws/datadog
    tags:
    - hostTag1
    - hostTag2
  override:
    clusterAgent:
      extraConfd:
        configMap:
          items:
          - key: test
            path: test.d/test.yaml
          name: cluster-agent-confd
      replicas: 2
      securityContext:
        runAsUser: 0
        seLinuxOptions:
          level: s0
          role: system_r
          type: spc_t
          user: system_u
    clusterChecksRunner:
      affinity:
        nodeAffinity:
          requiredDuringSchedulingIgnoredDuringExecution:
            nodeSelectorTerms:
            - matchExpressions:
              - key: kubernetes.io/os
                operator: In
                values:
                - linux
      annotations:
        baz: foo
        foo: bar
      createRbac: true
      customConfigurations:
        datadog.yaml:
          configData: custom-conf-clc
      image:
        jmxEnabled: true
        name: myagent
        pullPolicy: Always
        tag: mytag
      labels:
        bar: foo
        foo: baz
      name: test-clc-name
      nodeSelector:
        nodeName: clc
      priorityClassName: clc-prio-class
      replicas: 10
      securityContext:
        runAsUser: 0
        seLinuxOptions:
          level: s0
          role: system_r
          type: spc_t
          user: system_u
      serviceAccountName: datadog-agent-clc
      tolerations:
      - operator: Foo
    nodeAgent:
      annotations:
        baz: foo
        foo: bar
      containers:
        agent:
          args:
          - config
          - foo.yaml
          command:
          - custom-agent
          - start
          env:
          - name: TEST_VAR_CONT
            value: TEST_VALUE_CONT
          healthPort: 42
          livenessProbe:
            initialDelaySeconds: 30
          readinessProbe:
            timeoutSeconds: 10
          resources:
            limits:
              cpu: 400m
          volumeMounts:
          - mountPath: ""
            name: agent-volumeMount
      createRbac: true
      env:
      - name: TEST_VAR_DS
        value: TEST_VALUE_DS
      extraChecksd:
        configMap:
          name: agent-checksd
      extraConfd:
        configMap:
          name: agent-confd
      hostPID: true
      image:
        jmxEnabled: true
        name: myagent
        pullPolicy: Always
        tag: mytag
      labels:
        bar: foo
        foo: baz
      name: test-ds-agent
      securityContext:
        runAsUser: 0
        seLinuxOptions:
          level: s0
          role: system_r
          type: spc_t
          user: system_u
      serviceAccountName: datadog-agent-scc
      tolerations:
      - operator: Exists
      volumes:
      - name: agent-volume
status: {}
